website/blog/2025-02-07-best-alternative-to-mongodb-data-api-and-http-endpoints.md

```diff
@@ -0,0 +1,199 @@
+
+## FerretDB Data API
+
+FerretDB is an open-source MongoDB-compatible database that runs on PostgreSQL.
+With MongoDB's Data API deprecated, we saw the need for a new solution that offers similar functionality while remaining open and accessible.
+
+FerretDB's Data API provides:
+
+- Seamless HTTP-based database access – No need for drivers; interact via simple REST requests.
+- MongoDB-compatible queries – Use familiar JSON-based queries and commands.
+- PostgreSQL-backed storage – Leverage the reliability of PostgreSQL with a MongoDB-like API.
+
+To access Data API on FerretDB, set the environment variable or flag (`FERRETDB_LISTEN_DATA_API_ADDR`/`--listen-data-api-addr`) to your desired port when starting FerretDB.
+
+## Setting up FerretDB Data API
+
+### Get your FerretDB instance running
+
+You can set it up via Docker compose.
+Here's an example:
+
+```yaml
+services:
+  postgres:
+    image: ghcr.io/ferretdb/postgres-documentdb:16.6-0.100-0
+    restart: on-failure
+    environment:
+      - POSTGRES_USER=username
+      - POSTGRES_PASSWORD=password
+      - POSTGRES_DB=postgres
+    volumes:
+      - ./data:/var/lib/postgresql/data
+
+  ferretdb:
+    image: ghcr.io/ferretdb/ferretdb:2.0.0-rc.1
+    restart: on-failure
+    ports:
+      - 27017:27017
+      - 8080:8080
+    environment:
+      - FERRETDB_POSTGRESQL_URL=postgres://username:password@postgres:5432/postgres
+      - FERRETDB_LISTEN_DATA_API_ADDR=:8080
+networks:
+  default:
+    name: ferretdb
+```
+
+Note that the FerretDB v2 release requires a PostgreSQL with DocumentDB extension as the backend.
+Once your FerretDB instance is running, the Data API endpoint will be available at `http://localhost:8080`.
+
+### Using FerretDB's Data API for CRUD operations
+
+Since you've enabled FerretDB's Data API (`FERRETDB_LISTEN_DATA_API_ADDR=:8080`), let's write, query, update, and delete a single document using `curl`.
+
+#### Insert a document
+
+Insert a document into a collection called users in a database called `test`.
+
+```sh
+curl -X POST http://127.0.0.1:8080/action/insertOne \
+  -H "Content-Type: application/json" \
+  -u username:password \
+  -d '{
+        "database": "test",
+        "collection": "users",
+        "document": {
+          "name": "Andrew",
+          "email": "andrew@example.com",
+          "age": 25
+        }
+      }'
+```
+
+Response:
+
+```json
+{ "n": 1.0 }
+```
+
+#### Query data
+
+Next, let's retrieve the document we just inserted.
+
+```sh
+curl -X POST http://127.0.0.1:8080/action/find \
+  -H "Content-Type: application/json" \
+  -u username:password \
+  -d '{
+        "database": "test",
+        "collection": "users",
+        "filter": { "name": "Andrew" }
+      }'
+```
+
+Response:
+
+```json
+{
+  "documents": [
+    {
+      "_id": { "$oid": "67a2e9b70a5e9467a00918b0" },
+      "name": "Andrew",
+      "email": "andrew@example.com",
+      "age": 25
+    }
+  ]
+}
+```
+
+#### Update a document
+
+Let's update Alice's email:
```

Comment:
It's Andrew :hugs:, could you read though to check the mismatch please :pray:

Reply:
Thanks for spotting that 🙏🏽